DataAccessLibary/ReservationRepository.cs

```diff
@@ -202,4 +259,36 @@ public List<Reservation> GetReservations(Member member)
 
         return reservations.OrderBy(x => x.Id).ToList();
     }
+
+    public Reservation GetReservation(int reservationid)
+    {
+        Reservation reservations;
+        using (MySqlConnection connection = new MySqlConnection(ConnectionString.GetString()))
+        {
+            connection.Open();
+            const string sql = $"SELECT * FROM reservation WHERE `reservation_id` = @reservation_id";
+
+            using (MySqlCommand command = new MySqlCommand(sql, connection))
+            {
+                command.Parameters.Add("@reservation_id", MySqlDbType.Int32);
+                command.Parameters["@reservation_id"].Value = reservationid;
+
+                using (MySqlDataReader reader = command.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        var id = reader.GetInt32(0);
+                        var boatId = reader.GetInt32(1);
+                        var memberId = reader.GetInt32(2);
+                        var creationDate = reader.GetDateTime(3);
+                        var startTime = reader.GetDateTime(4);
+                        var endTime = reader.GetDateTime(5);
+
+                        return reservations = new Reservation(memberId, boatId, startTime, endTime);
```

Comment:
Je moet hier de constructor public Reservation(Member member, DateTime creationDate, DateTime startTime, DateTime endTime, int boatId, int id) gebruiken omdat je alle data hebt. Die constructor met 4 parameters wordt alleen in testen gebruikt en kan dus eigenlijk ook weg. Ook de lege constructor wordt niet gebruikt.